apis/v1alpha2/validation/gateway.go

```diff
@@ -29,49 +25,11 @@ import (
 // ValidateGateway validates gw according to the Gateway API specification.
 // For additional details of the Gateway spec, refer to:
 //  https://gateway-api.sigs.k8s.io/spec/#gateway.networking.k8s.io/v1alpha2.Gateway
+//
+// This function is currently empty as the validations have been moved to CRD field validations.
+// It's expected that further validation will be added back here in the future.
+// Here is the related PR for the reference:
+//  https://github.com/kubernetes-sigs/gateway-api/pull/872
```

Comment:
Maybe just something like this:

```suggestion
// Validation that is not possible with CRD annotations may be added here in the future.
// See https://github.com/kubernetes-sigs/gateway-api/issues/868 for more information.
```